File: server/src/test/java/com/pass/unit/exam/command/domain/QuestionOptionTest.java
package com.pass.unit.exam.command.domain;

import static org.assertj.core.api.Assertions.assertThatCode;
import static org.assertj.core.api.Assertions.assertThatThrownBy;

import com.pass.exam.command.domain.QuestionOption;
import com.pass.global.exception.BadRequestException;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.ValueSource;

class QuestionOptionTest {

    @Test
    @DisplayName("질문 옵션을 생성할 수 있다.")
    void create() {
        // when & then
        assertThatCode(() -> new QuestionOption("옵션 1", true))
                .doesNotThrowAnyException();
    }

    @ParameterizedTest
    @ValueSource(strings = {"", " "})
    @DisplayName("옵션 텍스트가 공백일 경우 예외를 발생시킨다.")
    void failWhenTextIsBlank(String text) {
        // when & then
        assertThatCode(() -> new QuestionOption(text, true))
                .isInstanceOf(BadRequestException.class)
                .hasMessage("질문 옵션의 길이는 1~200자 이어야 합니다.");
    }

    @ParameterizedTest
    @ValueSource(ints = {0, 201})
    @DisplayName("옵션 텍스트의 길이가 1~200자가 아닌 경우 경우 예외를 발생시킨다.")
    void failWhenTextLengthIsInvalid(int length) {
        // when
        String text = "a".repeat(length);

        // then
        assertThatThrownBy(() -> new QuestionOption(text, true))
                .isInstanceOf(BadRequestException.class)
                .hasMessage("질문 옵션의 길이는 1~200자 이어야 합니다.");
    }
}

File: server/src/test/kotlin/com/fluffy/exam/domain/QuestionOptionTest.kt
package com.fluffy.exam.domain

import com.fluffy.global.exception.BadRequestException
import io.kotest.assertions.throwables.shouldThrow
import io.kotest.core.spec.style.StringSpec
import io.kotest.data.forAll
import io.kotest.data.row
import io.kotest.matchers.shouldBe

class QuestionOptionTest : StringSpec({

    "문제 선택지가 정상적으로 생성된다." {
        // given
        val text = "a".repeat(200)
        val isCorrect = true

        // when
        val questionOption = QuestionOption(text, isCorrect)

        // then
        questionOption.text shouldBe text
    }

    "문제 선택지의 내용은 비어있을 수 없다." {
        forAll(
            row(""),
            row(" ")
        ) {
            shouldThrow<BadRequestException> { QuestionOption(it, true) }
                .message shouldBe "문제 선택지의 내용은 비어있을 수 없습니다."
        }
    }

    "문제 선택지의 내용은 200자 이하여야 한다." {
        // given
        val text = "a".repeat(201)

        // when & then
        shouldThrow<BadRequestException> { QuestionOption(text, true) }
            .message shouldBe "문제 선택지의 내용은 200자 이하여야 합니다."
    }
})
